codectrl-server/src/lib.rs

```diff
@@ -500,6 +507,49 @@ impl Authentication for Service {
     ) -> Result<Response<Token>, Status> {
         todo!()
     }
+
+    async fn github_login(
+        &self,
+        _request: Request<GhLogin>,
+    ) -> Result<Response<LoginUrl>, Status> {
+        Ok(Response::new(LoginUrl {
+            url: generate_github_login_url(),
+        }))
+    }
+}
+
+fn generate_github_login_url() -> String {
+    let github_client_id = ClientId::new(
+        env::var("GITHUB_CLIENT_ID")
+            .expect("Missing the GITHUB_CLIENT_ID environment variable."),
+    );
+    let github_client_secret = ClientSecret::new(
+        env::var("GITHUB_CLIENT_SECRET")
+            .expect("Missing the GITHUB_CLIENT_SECRET environment variable."),
```

Comment:
These checks should probably go down in the `run_server` function. We would want to warn about these being missing as early as possible rather than crashing out the server the first time when tries to login with GitHub.